app/controllers/mentor/regional_pitch_event_selections_controller.rb

```
@@ -8,10 +8,15 @@ def new
     def create
       event = RegionalPitchEvent.find(params[:event_id])
 
-      AddTeamToRegionalEvent.(event, current_team)
+      if event.at_team_capacity?
```

This is somewhat of a redundant check since `AddTeamToRegionalEvent` has it's own check. I don't necessarily see that as a problem, but there are 2 failure paths based on capacity: this check and the subsequent redirect, or the less likely (but I think still possible) scenario where this check passes but adding to the team actually fails due to race conditions, which I assume would result in a 500 since `AddTeamToRegionalEvent` raises an error?

---

The reason why I wanted this one here is so that the user sees some sort of feedback instead of an uh-oh page. I suppose this check is kind of unnecessary. I am open to leaving this or removing it. I leave that to you. If you would rather us fail hard with the `AddTeamToRegionalEvent` check, that is fine by me. I suppose this is a philosophical question then. Do we want to just fail hard in our code, or is feedback important with this sort of thing?

---

Right, I agree feedback is a good thing. Maybe do a try/catch catching the error that comes out of `AddTeamToRegionalEvent` and do the redirect there?

---

@stenington, after looking at this more, I am not sure how to redirect within that `AddTeamToRegionalEvent` class. I tried catching the error, but once I get into the rescue block, I don't have access to the route URL variables or any of the variables that would allow me to figure out what scope I am currently logged in as (student versus mentor). All of this makes routing at that level difficult. If you know of a straightforward way do that, perhaps we could look at this together, or you could take a crack at it and push to this PR. Otherwise, I might leave this functionality as is.